File: euler_065.py
from time import time


def a(n):
    if n == 0:
        return 2
    if n == 1:
        return 1
    if (n - 2)%3 == 0:
        return (2 * n + 2) // 3
    else:
        return 1


start = time()

a_list = [a(n) for n in range(100)]

p_list = [a_list[0], a_list[1]*a_list[0] + 1]
for n in range(2,100):
    p_list.append(a_list[n] * p_list[n - 1] + p_list[n - 2])

num = p_list[-1]
s = sum([int(d) for d in str(num)])

print("Sum of the digits of the numerator: " + str(s))

stop = time()
print("Time: " + str(stop-start))

File: euler_065.js
import { continuedFractionGeneratorSteps } from "./continuedFractions.mjs";
import Decimal from "./decimal.mjs";

Decimal.set({precision: 1000});

const start = Date.now();

const e = Decimal.exp(1);

const num = continuedFractionGeneratorSteps(e, 100)[1][99];
const sum = num.toString().split("").map(x => Number(x)).reduce((p,c) => p + c, 0);


const end = Date.now();

console.log("Execute time: ", end - start, " Sum of digits is: ", sum);